File: scripts/fix_yaml_comments/fix_yaml_comments.go
package fix_yaml_comments

import (
	"bufio"
	"bytes"
	"fmt"
	"os"
	"path/filepath"
	"regexp"
	"strings"
)

func isYAMLFile(firstLine string) bool {
	yamlStarters := []string{"---", "name:", "on:"}
	for _, starter := range yamlStarters {
		if strings.HasPrefix(firstLine, starter) {
			return true
		}
	}
	return false
}

func fixYAMLComments(filename string) (bool, error) {
	// Skip markdown files
	if ext := filepath.Ext(filename); ext == ".md" || ext == ".markdown" {
		return false, nil
	}

	content, err := os.ReadFile(filename)
	if err != nil {
		return false, err
	}

	// Check if file looks like YAML
	scanner := bufio.NewScanner(bytes.NewReader(content))
	if !scanner.Scan() || !isYAMLFile(scanner.Text()) {
		return false, nil
	}

	// Fix: Simpler regex that matches single # at start of line
	re := regexp.MustCompile(`(?m)^#([^#])`)
	modified := re.ReplaceAllString(string(content), "##$1")

	if string(content) != modified {
		err = os.WriteFile(filename, []byte(modified), 0644)
		if err != nil {
			return false, err
		}
		return true, nil
	}

	return false, nil
}

func FixYAMLComments(filenames ...string) error {
	if len(filenames) == 0 {
		return fmt.Errorf("no files provided")
	}

	changed := false
	for _, filename := range filenames {
		wasChanged, err := fixYAMLComments(filename)
		if err != nil {
			return fmt.Errorf("error processing %s: %v", filename, err)
		}
		if wasChanged {
			fmt.Printf("Fixed comments in %s\n", filename)
			changed = true
		}
	}

	if changed {
		return fmt.Errorf("files were modified")
	}
	return nil
}

File: scripts/fix_yaml_comments/fix_yaml_comments.py
#!/usr/bin/env python3
import re
import sys
from pathlib import Path

def fix_yaml_comments(file_path):
    # Skip markdown files
    if file_path.suffix.lower() in ['.md', '.markdown']:
        return False
    
    with open(file_path, 'r') as f:
        content = f.read()
    
    # Only process files that look like YAML
    if not any(content.startswith(prefix) for prefix in ['---', 'name:', 'on:']):
        return False

    # Replace single # comments with ## but ignore if already ##
    modified_content = re.sub(r'(?m)^([^#]*)(?<![#])#(?!#)\s', r'\1## ', content)
    
    if modified_content != content:
        with open(file_path, 'w') as f:
            f.write(modified_content)
        return True
    
    return False

def main():
    changed = False
    for file_path in sys.argv[1:]:
        if fix_yaml_comments(Path(file_path)):
            print(f"Fixed comments in {file_path}")
            changed = True
    
    sys.exit(1 if changed else 0)

if __name__ == '__main__':
    main() 